### DISCUSSION TITLE
Markitdown produces nothing?

### DISCUSSION THREAD

**Discussion by zhangguochun** [microsoft/markitdown#184] [Q&A]
Hello, does anyone have the same issue or know why. I can't use markitdown to covert a pdf. it doesn't throw any exceptions either. It just outputs blank. Thank you for any clues.

```python
from markitdown import MarkItDown
from langchain_ollama import ChatOllama
client = ChatOllama(model="llama3.2")
# md = MarkItDown(llm_client=client)
md = MarkItDown()
result = md.convert("test.pdf")
print(result.text_content)```

or tested it via command:
markitdown test.pdf

-----
markitdown = "^0.0.1a3"
WSL

**Comment by zhangguochun** [comment]
The script is correct. The PDF file "test.pdf" is a pure picture PDF file. When a text PDF is tested, it's good.

**Reply by timkinnane** [reply]
Is it possible to force parsing a pure picture PDF as an image file, instead of it being read as PDF from the file type and not being able to produce anything? I would essentially like to use `markitdown` for OCR on a PDF as if it were an image.

**Reply by mecqmx** [reply]
He would need integration with Azure Document Intelligence.  https://realpython.com/python-markitdown/

**Comment by arunbugkiller** [comment]
@zhangguochun were you able to resolve this issue? I am facing the same issue wherein result is a blank file.

**Reply by mecqmx** [reply]
He said it was because the PDF was scanned, so maybe your PDF is scanned or protected. In both cases, the output would be an empty PDF file.

**Comment by txhno** [comment]
Most likely the PDF has little or no extractable text.

Plain MarkItDown works best on text PDFs. If the file is basically scanned images, blank output is expected.

In that case you would need an OCR path, for example the `markitdown-ocr` plugin or Azure Document Intelligence.

**Comment by VANDRANKI** [comment]
If the PDF is scanned (just images of pages), markitdown will produce empty or minimal output because there is no embedded text to extract. pdfminer, which markitdown uses under the hood, only reads text that is actually encoded in the PDF.

To get text from a scanned PDF you need OCR. You can run the PDF through something like tesseract or a cloud OCR service first, then pass the resulting text or a searchable PDF back through markitdown. There is also ongoing discussion about adding an OCR path directly, but for now that pre-processing step is needed.
